gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/ImmunizationMapper.java

```diff
@@ -0,0 +1,171 @@
+        } else if (observationStatement.getEffectiveTime().getLow() != null) {
+            return DateFormatUtil.parse(observationStatement.getEffectiveTime().getLow().getValue()).getValue();
+        } else if (observationStatement.getEffectiveTime().getHigh() != null
+            && observationStatement.getAvailabilityTime() != null
+            && observationStatement.getAvailabilityTime().getValue() != null) {
+            return DateFormatUtil.parse(observationStatement.getAvailabilityTime().getValue()).getValue();
+        } else if (observationStatement.getEffectiveTime() == null && observationStatement.getAvailabilityTime().getValue() != null) {
+            return DateFormatUtil.parse(observationStatement.getAvailabilityTime().getValue()).getValue();
+        }
+
+        return null;
+    }
+
+    private String highValueToNotes(RCMRMT030101UK04ObservationStatement observationStatement) {
+        if (observationStatement.getEffectiveTime().getLow() != null && observationStatement.getEffectiveTime().getHigh() != null) {
+            return observationStatement.getEffectiveTime().getHigh().getValue();
+        } else if (observationStatement.getEffectiveTime().getHigh() != null
+            && observationStatement.getAvailabilityTime().getValue() != null) {
+            return observationStatement.getEffectiveTime().getHigh().getValue();
+        } else if (observationStatement.getEffectiveTime().getHigh() != null
+            && observationStatement.getAvailabilityTime().getNullFlavor() != null) {
+            return observationStatement.getEffectiveTime().getHigh().getValue();
+        }
+        return null;
+    }
+
+    private List<Annotation> buildNotes(Optional<RCMRMT030101UK04ObservationStatement> observationStatement) {
+        List<Annotation> annotationList = new ArrayList<>();
+
+        observationStatement.ifPresent(observationStatement1 -> observationStatement1.getPertinentInformation()
+            .stream()
+            .map(RCMRMT030101UK04PertinentInformation02::getPertinentAnnotation)
+            .filter(Objects::nonNull)
+            .map(RCMRMT030101UK04Annotation::getText)
+            .filter(StringUtils::isNotBlank)
+            .map(StringType::new)
+            .map(Annotation::new)
+            .forEach(annotationList::add));
+
+        if (observationStatement.get().getCode().getOriginalText() != null) {
+            StringType text = new StringType(observationStatement.get().getCode().getOriginalText());
+            annotationList.add(new Annotation(text));
+        }
+
+        return annotationList;
+    }
+
+    private List<Immunization> createImmunization(String id, Identifier identifier, List<Annotation> note,
+        Extension vaccineExtension, Date date, Extension recordedTimeExtension) {
+        var immunization = new Immunization();
+
+        immunization.getMeta().getProfile().add(new UriType(META_PROFILE));
+        immunization.getIdentifier().add(identifier);
+
+        try {
```

Comment:
We should't use kind of exception driven approach. I'm not really sure what are you trying to achieve here - definitely needs rework.